runtime/oti/j9nonbuilder.h

```diff
@@ -5000,6 +5000,10 @@ typedef struct J9VMThread {
 	struct J9GSParameters gsParameters;
 	void* evacuateBase;
 	void* evacuateTop;
+#if defined(J9VM_GC_COMPRESSED_POINTERS)
+	U_32 evacuateBaseCompressed;
+	U_32 evacuateTopCompressed;
+#endif /* J9VM_GC_COMPRESSED_POINTERS */
```

Comment:
This isn't the way I exactly imagined, but this could work, too. Like this, the code generation needs to check if it generates the code for 64bit CR or nonCR and use one set of bounds or the other (which it isn't as I can see - it just uses the CR set of bounds).

BTW I imagined that that even evacuateBase(Top)Compressed would also be 64bit (basically it would be of type fj9object_t *) and would be used for code that deals with heap references (that could be, but not necessarily are compressed, like if we run 64bit nonCR), and the original set of evacuateBase/Top (basically of type j9object_t *) would be used with code that deal with off heap structures, locals, native method arguments etc, that are known to deal with already uncompressed form of a heap reference.

Reply:
Not clear to me why we need separate variables for compressed and not - only one can be true in any given build.

Reply:
I see your point, but IMO for nonCR build, there is no need to keep two identical ranges. The JIT evaluator need a separate evaluator for nonCR build anyway, so that I don't see the advantage keep two identical ranges on nonCR.

Reply:
In CR build we need both forms (CR and nonCR) for the ranges. When jitted code wants to check a range for a reference just read from an object in heap it wants to compare it with compressed from of ranges .
If jitted code (or any other VM code) wants to check a range for reference just read (for example) from class static it wants to compare it with non-compressed form of ranges.

Reply:
@0dvictor Yes, understand your point. A possible advantage of the other approach is to avoid runtime checkes in non generated code (when running CR) that need to distinguish between the two set of ranges. Maybe there are not many (or at all, now), but in future CR and non-CR can actually get merged.

I don't have a strong preference, both approaches are ok to me. If someone has stronger preference let's hear it.